File: 1983/1983.go
func widestPairOfIndicesWithEqualRa(root *TreeNode) int {
	if root == nil {
		return 0
	}
	l := widestPairOfIndicesWithEqualRa(root.Left)
	r := widestPairOfIndicesWithEqualRa(root.Right)
	if l > r {
		return 1 + l
	}
	return 1 + r
}

File: 1983/1983.cpp
class Solution {
public:
    int widestPairOfIndicesWithEqualRa(TreeNode* root) {
        if(!root) return 0;
        return 1+max(widestPairOfIndicesWithEqualRa(root->left),widestPairOfIndicesWithEqualRa(root->right));
    }
};